test/bats/provider.bats

```diff
@@ -63,92 +78,141 @@ teardown(){
-  result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store-2/bar2)
-  [[ "$result" == "hello-sync2" ]]
-
-  result=$(kubectl --namespace=test get secret foosecret-1 -o jsonpath="{.data.username}" | base64 -d)
-  [[ "$result" == "hello-sync1" ]]
-  result=$(kubectl --namespace=test get secret foosecret-2 -o jsonpath="{.data.pwd}" | base64 -d)
-  [[ "$result" == "hello-sync2" ]]
-
-  result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_1_USERNAME | awk -F"=" '{ print $2 }' | tr -d '\r\n')
-  [[ "$result" == "hello-sync1" ]]
-  result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_2_PWD | awk -F"=" '{ print $2 }' | tr -d '\r\n')
-  [[ "$result" == "hello-sync2" ]]
+    POD=nginx-multiple-volumes
+    kubectl --namespace=test apply -f $CONFIGS/nginx-multiple-volumes.yaml
+    kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod $POD
+
+    result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store-1/secret-1)
+    [[ "$result" == "hello-sync1" ]]
+    result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store-2/secret-2)
+    [[ "$result" == "hello-sync2" ]]
+
+    result=$(kubectl --namespace=test get secret foosecret-1 -o jsonpath="{.data.username}" | base64 -d)
+    [[ "$result" == "hello-sync1" ]]
+    result=$(kubectl --namespace=test get secret foosecret-2 -o jsonpath="{.data.pwd}" | base64 -d)
+    [[ "$result" == "hello-sync2" ]]
+
+    result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_1_USERNAME | awk -F"=" '{ print $2 }' | tr -d '\r\n')
+    [[ "$result" == "hello-sync1" ]]
+    result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_2_PWD | awk -F"=" '{ print $2 }' | tr -d '\r\n')
+    [[ "$result" == "hello-sync2" ]]
+}
+
+@test "5 SecretProviderClass with query parameters and PUT method" {
+    kubectl --namespace=test apply -f $CONFIGS/nginx-pki.yaml
+    kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod nginx-pki
+
+    result=$(kubectl --namespace=test exec nginx-pki -- cat /mnt/secrets-store/certs)
+    [[ "$result" != "" ]]
+
+    # Ensure we have some valid x509 certificates.
+    echo "$result" | jq -r '.data.certificate' | openssl x509 -noout
+    echo "$result" | jq -r '.data.issuing_ca' | openssl x509 -noout
+    echo "$result" | jq -r '.data.certificate' | openssl x509 -noout -text | grep "test.example.com"
+}
+
+@test "6 Dynamic secrets engine, endpoint is called only once per SecretProviderClass" {
+    # Setup postgres
+    POSTGRES_PASSWORD=$(openssl rand -base64 30)
+    kubectl --namespace=test create secret generic postgres-root \
+        --from-literal=POSTGRES_USER="root" \
+        --from-literal=POSTGRES_PASSWORD="${POSTGRES_PASSWORD}"
+    kubectl --namespace=test apply -f $CONFIGS/postgres.yaml
+    kubectl wait --namespace=test --for=condition=Ready --timeout=5m pod postgres
+    POSTGRES_POD_IP=$(kubectl --namespace=test get pod postgres -o json | jq -r '.status.podIP')
```

Comment:
This could be simplified by connecting to the PostgreSQL service instead. Pod IPs are usually not used.

Reply:
Yeah I initially looked down the route of using a service but decided it wasn't worth the extra config vs a call to find the pod IP. To go the service route, we need to turn the raw pod into a deployment, create the service with the selector labels, and then we still need to query with `kubectl` for some info about the pod, but this time for the pod's name so we can `exec` instead of for the pod's IP. Overall it seems simpler just to query the IP. WDYT?

Reply:
Of course that's not what you'd do outside of a test, but seems ok in this very carefully curated test environment.

Reply:
Just to clarify, a pod can have a service without a deployment. Services work via labels and as long as the pod matches the labels, the service will route to it.

Reply:
Good improvement 👍 done in the next PR